packages/replay/src/util/addGlobalListeners.ts

```diff
@@ -23,7 +27,15 @@ export function addGlobalListeners(replay: ReplayContainer): void {
   addInstrumentationHandler('xhr', handleXhrSpanListener(replay));
   addInstrumentationHandler('history', handleHistorySpanListener(replay));
 
+  // If a custom client has no hooks yet, we continue to use the "old" implementation
+  const hasHooks = !!(client && client.on);
+
   // Tag all (non replay) events that get sent to Sentry with the current
   // replay ID so that we can reference them later in the UI
-  addGlobalEventProcessor(handleGlobalEventListener(replay));
+  addGlobalEventProcessor(handleGlobalEventListener(replay, !hasHooks));
+
+  if (hasHooks) {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (client as BaseClient<any>).on('afterSendErrorEvent', handleAfterSendError(replay));
```

Comment:
should this be a more general hook that is called `afterSendEvent`? Then in `handleAfterSendError` we can check for `isErrorEvent`.

Reply:
Yeah, I wasn't entirely sure here 🤔 I can make it more generic. My main reasoning was to avoid the `beforeSend`/`beforeSendTransaction` naming mess we have, down the line 😅

Reply:
+1 for generic -- we'll want to do something similar with `traceIds`, no?

Reply:
I will make it generic, but please note that this has downsides to - as `Event` can also be a replay event or profile event, we'll have to guard this everywhere etc., even though _actually_ as of now these events go through a different code path and can't even get in there.